### PR TITLE
refactor(cache): use a single cache for all image storages

### PR DESCRIPTION
**Pull request by peusebiu**
Signed-off-by: Petu Eusebiu <peusebiu@cisco.com>

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/storage/local/local_test.go [hunk 1]
```diff
@@ -40,18 +40,35 @@ const (
...
-	cacheDriver, _ := storage.Create("boltdb", cache.BoltDBDriverParameters{
-		RootDir:     dir,
-		Name:        "cache",
-		UseRelPaths: true,
+
+	var cacheDriver cache.Cache
+
+	cacheDriver, _ = storage.Create("boltdb", cache.BoltDBDriverParameters{
+		RootDir: rootDir,
+		Name:    "s3_cache",
```
Comment (andaaron) [inline pkg/storage/local/local_test.go:51]:
```suggestion
		Name:    "cache",
```

**Review by andaaron — COMMENTED** [review]
Remove pkg/zot, I think it was unintentionally added to the git tree.

**Comment by rchincha** [comment]
@peusebiu can you rebase and resubmit pls.

File: examples/config-dynamodb.json [hunk 2]
```diff
@@ -17,7 +16,6 @@
             "name": "dynamodb",
             "endpoint": "http://localhost:4566",
             "region": "us-east-2",
-            "cacheTablename": "ZotBlobTable",
```
[the remark this answers is not in the document]
Comment (peusebiu) [reply examples/config-dynamodb.json:20]: Added tableNamePrefix

File: pkg/cli/root.go [hunk 3]
```diff
@@ -238,24 +235,7 @@ func validateStorageConfig(cfg *config.Config) error {
...
-		// local storage with remote caching
-		if cfg.Storage.StorageDriver == nil {
-			log.Error().Err(errors.ErrBadConfig).Msg("cannot have local storage driver with remote caching!")
-
-			return errors.ErrBadConfig
-		}
-
+	if cfg.Storage.CacheDriver != nil {
 		// unsupported cache driver
 		if cfg.Storage.CacheDriver["name"] != storageConstants.DynamoDBDriverName {
```
[the remark this answers is not in the document]
Comment (peusebiu) [reply pkg/cli/root.go:240]: Let's do it in a future patch, too many changes needed.

File: pkg/api/controller.go [hunk 4]
```diff
@@ -276,14 +280,34 @@ func (c *Controller) InitImageStore(ctx context.Context) error {
 			}
 		}
 
+		if hasDedupe(c.Config.Storage) {
```
[the remark this answers is not in the document]
Comment (peusebiu) [reply pkg/api/controller.go]: done

File: pkg/cli/root.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (peusebiu) [reply pkg/cli/root.go:240]: let's do it in a future patch, there are too many changes in this one

[the remark this answers is not in the document]
Comment (andaaron) [reply pkg/cli/root.go:240]: Is it more thank validating cfg.Storage.CacheDriver["name"] is one of storageConstants.DynamoDBDriverName or an equivalent for BoltDB?

[the remark this answers is not in the document]
Comment (peusebiu) [reply pkg/cli/root.go:240]: yeah, there are a lot of changes needed, I will do it, but it will break some tests.

**Comment by andaaron** [comment]
This PR is too old to be maintained. Let's open a new one if needed.

**Pull request closed without merging**